### PR TITLE
fakeClient should always return NotFound error

### PR DESCRIPTION
**Pull request by justinsb**
Although it's a fake client, we want it to behave like the real one!

### CODE REVIEW

File: pkg/test/mocks/client_test.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+	client := NewClient(scheme)
+
+	node := v1.Node{}
+	err := client.Get(ctx, types.NamespacedName{Name: "foo"}, &node)
+	if !apierrors.IsNotFound(err) {
+		t.Fatalf("expected IsNotFound error from non-existent get, got %v", err)
+	}
+}
+
+func TestCreateAndGetGet(t *testing.T) {
```
Comment (atoato88) [inline pkg/test/mocks/client_test.go:46]: Is `Get` duplicated?

```suggestion
func TestCreateAndGet(t *testing.T) {
```

**Review by atoato88 — COMMENTED** [review]
Almost LGTM.
Just one comment. it may be nit. :thinking:

File: pkg/test/mocks/client_test.go (hunk 1 above)
Comment (justinsb) [reply pkg/test/mocks/client_test.go]: Whoops - thanks!

**Review by rajansandeep — APPROVED** [review]
/lgtm

**Pull request merged**